proto file(s) should be named with package name not service name (if package name defined)

When consuming two Goa gRPC services from a third service where the first two share a service name but have a distinct Package name defined in the design, we found that proto was panicking (we believe) because of the .proto files having the same filename in each service being consumed.

goa designs:

_system A_
```
var _ = Service("Tester", func() {
	GRPC(func() {
		Package("systema_tester")
	})
...
}
```

_system B_
```
var _ = Service("Tester", func() {
	GRPC(func() {
		Package("systemb_tester")
	})
...
}
```

The error seen when the consuming service is built or tested:
```
panic: proto: file "goadesign_goagen_tester.proto" is already registered
        previously from: "github.com/org/systemA/gen/grpc/tester/pb"
        currently from:  "github.com/org/systemB/gen/grpc/tester/pb"
See https://protobuf.dev/reference/go/faq#namespace-conflict
```

This seems to be because the .proto (and other pb files) are named after the gRPC _service name_, not after the _package name_.

I think [this line in proto.go](https://github.com/goadesign/goa/blob/3698a579e50a6b59a3c5cec654da99160e699ddf/grpc/codegen/proto.go#L31) is the culprit. Currently as written, it generates the filename as:
```
path := filepath.Join(codegen.Gendir, "grpc", svcName, pbPkgName, "goadesign_goagen_"+svcName+".proto")
```
where instead it could do something like:
```
path := filepath.Join(codegen.Gendir, "grpc", svcName, pbPkgName, "goadesign_goagen_"+pkgName(svc, svcName)+".proto")
```
as [func pkgName](https://github.com/goadesign/goa/blob/3698a579e50a6b59a3c5cec654da99160e699ddf/grpc/codegen/proto.go#L79) seems to take care of the logic that uses service name instead of package name if `svc.ProtoPkg` is not defined.

Unsure if other files generated in the `/gen/grpc/<servicename>/pb/` folder need to match this pattern as well (`goadesign_goagen_tester.pb.go` and `goadesign_goagen_tester_grpc.pb.go`)

The current workaround is to use the following flag on both `test` and `build` to suppress the error as a warning:
```
-ldflags "-X google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn"
```

While doable in the short term, not a fan of how that could mask real issues in the future so a change from Goa to address the filenames would be most helpful.

Thanks!

--- a/grpc/codegen/proto.go
+++ b/grpc/codegen/proto.go
@@ -5,6 +5,7 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
 	"goa.design/goa/v3/codegen"
 	"goa.design/goa/v3/expr"
@@ -14,6 +15,9 @@ import (
 const (
 	// ProtoVersion is the protocol buffer version used to generate .proto files
 	ProtoVersion = "proto3"
+
+	// ProtoPrefix is the prefix added to the proto package name.
+	ProtoPrefix = "goagen"
 )
 
 // ProtoFiles returns a *.proto file for each gRPC service.
@@ -28,7 +32,16 @@ func ProtoFiles(genpkg string, root *expr.RootExpr) []*codegen.File {
 func protoFile(genpkg string, svc *expr.GRPCServiceExpr) *codegen.File {
 	data := GRPCServices.Get(svc.Name())
 	svcName := data.Service.PathName
-	path := filepath.Join(codegen.Gendir, "grpc", svcName, pbPkgName, "goadesign_goagen_"+svcName+".proto")
+	parts := strings.Split(genpkg, "/")
+	var repoName string
+	if len(parts) > 1 {
+		repoName = parts[len(parts)-2]
+	} else {
+		repoName = parts[0]
+	}
+	// the filename is used by protoc to set the namespace so try to make it unique
+	fname := fmt.Sprintf("%s_%s_%s.proto", ProtoPrefix, repoName, svcName)
+	path := filepath.Join(codegen.Gendir, "grpc", svcName, pbPkgName, fname)
 
 	sections := []*codegen.SectionTemplate{
 		// header comments

